lib/graph/helpers/ui5Framework.js

```diff
@@ -192,6 +192,22 @@ const utils = {
 		});
 		await projectGraph.resolveOptionalDependencies();
 	},
+	checkForDuplicateFrameworkProjects(projectGraph, frameworkGraph) {
+		// Check for duplicate framework libraries
+		const projectGraphProjectNames = projectGraph.getProjectNames();
+		const duplicateFrameworkProjectNames = frameworkGraph.getProjectNames()
+			.filter((name) => projectGraphProjectNames.includes(name));
+
+		if (duplicateFrameworkProjectNames.length) {
```

Comment:
Maybe `Array.some()` would suit better here:

```suggestion
			.some((name) => projectGraphProjectNames.includes(name));

		if (duplicateFrameworkProjectNames) {
```